docs/accounting/Accounting.lhs

```diff
@@ -0,0 +1,405 @@
+  , _lastEpoch :: Slot }
+\end{code}
+%if style == code
+\begin{code}
+  deriving (Show, Eq)
+
+makeLenses ''PrtclConsts
+makeLenses ''LedgerState
+
+\end{code}
+%endif
+
+The $\mathsf{\_lastEpoch}$ variable is used only
+to track when decayed portions of a deposit
+are moved from the deposit pool to the reward pool,
+but for the purposes of this model we do not need to
+assuming a certain number of slots per epoch, etc.
+
+\subsection{Note on lens notation}
+
+We use the lens operators for getting and setting items
+in the ledger state and protocol constants.
+For example, to get the $\mathsf{\_minDep}$ from
+a protocol constants record, we write
+
+< pc ^. minDep
+
+And to get the $\mathsf{\_minDep}$ from a ledger state
+record, we write
+
+< ls ^. pconsts . minDep
+
+To set an element of a record, we write
+
+< ls & slot .~ Slot 42
+
+And to set multiple elements we write
+
+< ls & deposits .~ Coin 7
+<    & slot .~ Slot 42
+
+\subsection{Methods}
+
+The total amount of \lovelace in a given ledger state is
+given by the following calculation. It should always be
+equal to forty-five quadrillion \lovelace.
+\begin{code}
+total :: LedgerState -> Coin
+total ls = ls^.circulation + ls^.deposits + ls^.treasury
+            + ls^.reserves + ls^.rewards + ls^.rewardPool + ls^.fees
+
+\end{code}
+
+The deposit amounts are determined by the protocol constants.
+\begin{code}
+deposit :: Cert -> PrtclConsts -> Coin
+deposit (KeyReg _) = _keyRegDep
+deposit (PoolReg _) = _poolRegDep
+
+\end{code}
+
+The refund of a certificate is determined by an exponential decay.
+\begin{code}
+refund :: Cert -> PrtclConsts -> Duration -> Coin
+refund cert pc dur = floor (dep * (dmin + (1-dmin) * exp pow))
+  where
+    dep = fromIntegral $ deposit cert pc
+    dmin = pc^.minDep
+    pow = - pc^.decayRate * fromIntegral dur
+
+\end{code}
+
+For a given ledger state, we can calculate the total amount
+of \lovelace needed to refund everyone at the current slot.
+\begin{code}
+obligation :: LedgerState -> Coin
+obligation ls = sum
+  [refund cert (ls^.pconsts) ((ls^.slot) -* start)
+    | (cert, start) <- Map.toList (ls^.obligations)]
+
+\end{code}
+
+\subsection{Actions}
+
+There are seven types of actions that cause \lovelace
+to move between the seven categories.
+
+\begin{code}
+data Action = ActTxBody Coin
+            | ActAddCert Cert
+            | ActDelCert Cert
+            | ActWithdrawal Coin
+            | ActEpoch Float
+            | ActVote PrtclConsts
+            | ActNextSlot
+\end{code}
+%if style == code
+\begin{code}
+            deriving (Show, Eq)
+
+\end{code}
+%endif
+
+We now describe how each of these actions changes the
+ledger state.
+
+\begin{code}
+applyAction :: LedgerState -> Action -> LedgerState
+
+\end{code}
+
+The first actions is a UTxO transaction, which
+moves \lovelace out of circulation into the fee pool.
+In a real implementation, the validation rules would prevent
+a fee from beeing too high, but here we just skip it.
```

Comment:
I just meant that in the `applyAction` method, whenever there is a case that doesn't make sense, it just returns back the original ledger state instead of indicating some kind of validation error.  Like below, the case `fee > ls^.circulation` is just ignored.  Maybe this sentence is more confusing than helpful. :)